lib/SILGen/SILGenThunk.cpp

```diff
@@ -69,6 +69,51 @@ SILFunction *SILGenModule::getDynamicThunk(SILDeclRef constant,
   return F;
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+SILFunction *SILGenModule::getAutoDiffThunk(SILDeclRef constant,
+                                            CanSILFunctionType constantTy) {
+  auto *autoDiffFuncId = constant.autoDiffAssociatedFunctionIdentifier;
+  assert(autoDiffFuncId);
+
+  SILGenFunctionBuilder builder(*this);
+  auto originalFn = constant.asAutoDiffOriginalFunction();
+  auto originalLinkage = originalFn.getLinkage(ForDefinition);
+  auto linkage = autodiff::getAutoDiffAssociatedFunctionLinkage(
+      originalLinkage, /*isAssocFnExported*/ true);
+  auto name = constant.mangle() + "_thunk";
```

Comment:
Similar to above: consider generating a more specific thunk name since `getAutoDiffThunk` is currently only used for thunking class member JVPs/VJPs.